Added support to PIC18F4580 and possibility of multiple BRG configura…

nos86 opened the pull request:
…tion selected using _XTAL_FREQ definition

grodansparadis commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/grodansparadis/vscp_firmware/pull/6'>Refresh</a>.<a href='#crh-update-none'></a>

nos86 commented:
Hello,

I added the support to new Microchip devices PIC18F*580.

Moreover, I added the oppportunity to save several BRG configuration based on Crystal frequency than select it defining _XTAL_FREQ in vscp_projdefs.h

For backcompatibility, in case of mission definition, a warning is risen at compiling time and default values are used.

Best Regards,
Salvo

grodansparadis commented:
:shipit:

grodansparadis commented:
:shipit:

grodansparadis commented:
Thanks! Recommends ecan for new work. See https://github.com/grodansparadis/can4vscp_paris

grodansparadis commented:
:+1:
